[BUG] Build failure with only feature dns-over-https

`cargo check --no-default-features --features dns-over-https` failed with:

```
warning: /Users/nobodyxu/Dev/trust-dns/bin/Cargo.toml: file found to be present in multiple build targets: /Users/nobodyxu/Dev/trust-dns/bin/src/named.rs
    Checking trust-dns-proto v0.22.0 (/Users/nobodyxu/Dev/trust-dns/crates/proto)
error[E0433]: failed to resolve: use of undeclared crate or module `tokio_rustls`
  --> crates/proto/src/https/https_client_stream.rs:26:5
   |
26 | use tokio_rustls::{
   |     ^^^^^^^^^^^^ use of undeclared crate or module `tokio_rustls`

error[E0432]: unresolved import `rustls`
  --> crates/proto/src/https/https_client_stream.rs:25:5
   |
25 | use rustls::ClientConfig;
   |     ^^^^^^ use of undeclared crate or module `rustls`

error[E0432]: unresolved import `tokio_rustls`
  --> crates/proto/src/https/https_client_stream.rs:26:5
   |
26 | use tokio_rustls::{
   |     ^^^^^^^^^^^^ use of undeclared crate or module `tokio_rustls`

error[E0432]: unresolved import `crate::iocompat`
  --> crates/proto/src/https/https_client_stream.rs:32:12
   |
32 | use crate::iocompat::AsyncIoStdAsTokio;
   |            ^^^^^^^^ could not find `iocompat` in the crate root

error[E0433]: failed to resolve: use of undeclared crate or module `tokio`
   --> crates/proto/src/https/https_client_stream.rs:477:21
    |
477 |                     tokio::spawn(
    |                     ^^^^^ use of undeclared crate or module `tokio`

warning: unused import: `TryFutureExt`
  --> crates/proto/src/https/https_client_stream.rs:20:39
   |
20 | use futures_util::future::{FutureExt, TryFutureExt};
   |                                       ^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
warning: `trust-dns-proto` (lib) generated 1 warning
error: could not compile `trust-dns-proto` due to 5 previous errors; 1 warning emitted
```

I found this when running `cargo hack check --feature-powerset`.

**System:**
This bug does not depend on specific system configure to reproduce.

**Version:**
Crate: proto
Version: 0.22.0

**Additional context**

This bug is originally found in https://github.com/cargo-bins/cargo-binstall/pull/538 and https://github.com/cargo-bins/cargo-binstall/pull/430
It can be reproduced on GHA mac (x86-64)/windows/ubuntu whenever trust-dns is enabled and also on my M1 mac.

NOTE that enabling `dns-over-native-tls` also failed:

```
info: running `cargo check --no-default-features --features dns-over-native-tls` on trust-dns-proto (7/568
0)
warning: /Users/nobodyxu/Dev/trust-dns/bin/Cargo.toml: file found to be present in multiple build targets:
 /Users/nobodyxu/Dev/trust-dns/bin/src/named.rs
   Compiling tokio v1.21.0
    Checking tokio-native-tls v0.3.0
    Checking trust-dns-proto v0.22.0 (/Users/nobodyxu/Dev/trust-dns/crates/proto)
error[E0432]: unresolved import `crate::iocompat`
  --> crates/proto/src/native_tls/tls_client_stream.rs:21:12
   |
21 | use crate::iocompat::AsyncIoStdAsTokio;
   |            ^^^^^^^^ could not find `iocompat` in the crate root

error[E0432]: unresolved import `crate::iocompat`
  --> crates/proto/src/native_tls/tls_client_stream.rs:22:12
   |
22 | use crate::iocompat::AsyncIoTokioAsStd;
   |            ^^^^^^^^ could not find `iocompat` in the crate root

error[E0432]: unresolved import `crate::iocompat`
  --> crates/proto/src/native_tls/tls_stream.rs:20:12
   |
20 | use crate::iocompat::{AsyncIoStdAsTokio, AsyncIoTokioAsStd};
   |            ^^^^^^^^ could not find `iocompat` in the crate root

warning: unused import: `futures_util::TryFutureExt`
  --> crates/proto/src/native_tls/tls_client_stream.rs:14:5
   |
14 | use futures_util::TryFutureExt;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default


For more information about this error, try `rustc --explain E0432`.
warning: `trust-dns-proto` (lib) generated 1 warning
error: could not compile `trust-dns-proto` due to 3 previous errors; 1 warning emitted
error: process didn't exit successfully: `/Users/nobodyxu/.rustup/toolchains/stable-aarch64-apple-darwin/b
in/cargo check --manifest-path Cargo.toml --no-default-features --features dns-over-native-tls` (exit stat
us: 101)


```

--- a/crates/proto/Cargo.toml
+++ b/crates/proto/Cargo.toml
@@ -40,7 +40,7 @@ maintenance = { status = "actively-developed" }
 dns-over-tls = []
 dns-over-rustls = ["dns-over-tls", "rustls", "rustls-pemfile", "tokio-rustls", "webpki"]
 dns-over-native-tls = ["dns-over-tls", "native-tls", "tokio-native-tls", "tokio-runtime"]
-dns-over-openssl = ["dns-over-tls", "openssl", "tokio-openssl"]
+dns-over-openssl = ["dns-over-tls", "openssl", "tokio-openssl", "tokio-runtime"]
 
 dns-over-https-rustls = ["dns-over-https"]
 dns-over-https = ["bytes", "h2", "http", "dns-over-rustls", "webpki-roots", "tokio-runtime"]

--- a/crates/proto/Cargo.toml
+++ b/crates/proto/Cargo.toml
@@ -45,7 +45,7 @@ dns-over-openssl = ["dns-over-tls", "openssl", "tokio-openssl", "tokio-runtime"]
 dns-over-https-rustls = ["dns-over-https"]
 dns-over-https = ["bytes", "h2", "http", "dns-over-rustls", "webpki-roots", "tokio-runtime"]
 
-dns-over-quic = ["quinn", "rustls/quic", "dns-over-rustls", "bytes", "webpki-roots"]
+dns-over-quic = ["quinn", "rustls/quic", "dns-over-rustls", "bytes", "webpki-roots", "tokio-runtime"]
 
 dnssec-openssl = ["dnssec", "openssl"]
 dnssec-ring = ["dnssec", "ring"]

--- a/crates/proto/Cargo.toml
+++ b/crates/proto/Cargo.toml
@@ -42,8 +42,8 @@ dns-over-rustls = ["dns-over-tls", "rustls", "rustls-pemfile", "tokio-rustls", "
 dns-over-native-tls = ["dns-over-tls", "native-tls", "tokio-native-tls"]
 dns-over-openssl = ["dns-over-tls", "openssl", "tokio-openssl"]
 
-dns-over-https-rustls = ["dns-over-https", "dns-over-rustls", "webpki-roots"]
-dns-over-https = ["bytes", "dns-over-tls", "h2", "http"]
+dns-over-https-rustls = ["dns-over-https"]
+dns-over-https = ["bytes", "h2", "http", "dns-over-rustls", "webpki-roots", "tokio-runtime"]
 
 dns-over-quic = ["quinn", "rustls/quic", "dns-over-rustls", "bytes", "webpki-roots"]
 

--- a/crates/proto/Cargo.toml
+++ b/crates/proto/Cargo.toml
@@ -39,7 +39,7 @@ maintenance = { status = "actively-developed" }
 [features]
 dns-over-tls = []
 dns-over-rustls = ["dns-over-tls", "rustls", "rustls-pemfile", "tokio-rustls", "webpki"]
-dns-over-native-tls = ["dns-over-tls", "native-tls", "tokio-native-tls"]
+dns-over-native-tls = ["dns-over-tls", "native-tls", "tokio-native-tls", "tokio-runtime"]
 dns-over-openssl = ["dns-over-tls", "openssl", "tokio-openssl"]
 
 dns-over-https-rustls = ["dns-over-https"]

--- a/crates/proto/Cargo.toml
+++ b/crates/proto/Cargo.toml
@@ -38,7 +38,7 @@ maintenance = { status = "actively-developed" }
 
 [features]
 dns-over-tls = []
-dns-over-rustls = ["dns-over-tls", "rustls", "rustls-pemfile", "tokio-rustls", "webpki"]
+dns-over-rustls = ["dns-over-tls", "rustls", "rustls-pemfile", "tokio-rustls", "webpki", "tokio-runtime"]
 dns-over-native-tls = ["dns-over-tls", "native-tls", "tokio-native-tls", "tokio-runtime"]
 dns-over-openssl = ["dns-over-tls", "openssl", "tokio-openssl", "tokio-runtime"]
 

